scripts/measures/2018/import-quality-measures.js

```diff
@@ -0,0 +1,261 @@
+        'patient engagement/experience': 'patientEngagementExperience',
+        'efficiency': 'efficiency',
+        'intermediate outcome': 'intermediateOutcome',
+        'structure': 'structure',
+        'patient reported outcome': 'outcome',
+        'composite': 'outcome',
+        'cost/resource use': 'efficiency',
+        'clinical process effectiveness': 'process'
+      }
+    },
+    primarySteward: 9,
+    metricType: 51,
+    firstPerformanceYear: {
+      index: 52,
+      default: 2017
+    },
+    lastPerformanceYear: {
+      index: 53,
+      default: null
+    },
+    isHighPriority: {
+      index: 55,
+      default: false
+    },
+    isInverse: {
+      index: 56,
+      default: false
+    },
+    overallAlgorithm: 60
+  }
+};
+
+// mapping from quality measures csv column numbers to submission method
+const SUBMISSION_METHODS = {
+  10: 'claims',
+  11: 'certifiedSurveyVendor',
+  12: 'electronicHealthRecord',
+  13: 'cmsWebInterface',
+  14: 'administrativeClaims',
+  15: 'registry'
+};
+
+// mapping from quality measures csv column numbers to measure sets
+const MEASURE_SETS = {
+  16: 'allergyImmunology',
+  17: 'anesthesiology',
+  18: 'cardiology',
+  19: 'dermatology',
+  20: 'diagnosticRadiology',
+  21: 'electrophysiologyCardiacSpecialist',
+  22: 'emergencyMedicine',
+  23: 'gastroenterology',
+  24: 'generalOncology',
+  25: 'generalPracticeFamilyMedicine',
+  26: 'generalSurgery',
+  27: 'hospitalists',
+  28: 'internalMedicine',
+  29: 'interventionalRadiology',
+  30: 'mentalBehavioralHealth',
+  31: 'neurology',
+  32: 'obstetricsGynecology',
+  33: 'ophthalmology',
+  34: 'orthopedicSurgery',
+  35: 'otolaryngology',
+  36: 'pathology',
+  37: 'pediatrics',
+  38: 'physicalMedicine',
+  39: 'plasticSurgery',
+  40: 'preventiveMedicine',
+  41: 'radiationOncology',
+  42: 'rheumatology',
+  43: 'thoracicSurgery',
+  44: 'urology',
+  45: 'vascularSurgery'
+};
+
+function getCsv(csvPath, headerRows = 1) {
+  const csv = fs.readFileSync(path.join(__dirname, csvPath), 'utf8');
+  const parsedCsv = parse(csv, 'utf8');
+
+  // remove header rows
+  for (let i = 0; i < headerRows; i++) {
+    parsedCsv.shift();
+  }
+
+  return parsedCsv;
+}
+
+// Accounts for TRUE, True, true, X, x...
+// and people sometimes insert extra spaces
+function cleanInput(input) {
+  return input.trim().toLowerCase();
+}
+
+// map specific csv input values to their representation in the measures schema
+function mapInput(input) {
+  const cleanedInput = cleanInput(input);
+  if (cleanedInput === 'true' || cleanedInput === 'x') {
+    return true;
+  } else if (cleanedInput === 'false') {
+    return false;
+  } else if (cleanedInput === 'null' || cleanedInput === 'n/a') {
+    return null;
+  } else if (Constants.validPerformanceYears.includes(Number(cleanedInput))) {
+    return Number(cleanedInput);
+  } else {
+    // if csv input isn't one of the special cases above, just return it
+    return input.trim();
```

Comment:
no it hasn't, cleanedinput has been trimmed but we don't want to return cleanedinput because that's also lowercased, which we don't want (except when it's one of the special cases above this code)

Reply:
Oooh, duh!

Reply:
i'll rename input to rawInput